modules/avatar/react/lib/Avatar.tsx

```diff
@@ -10,80 +11,136 @@ export enum AvatarVariant {
   Dark,
 }
 
-export interface AvatarLocalProps {
+export interface AvatarProps extends React.HTMLAttributes<HTMLButtonElement> {
   /**
    * The variant of the Avatar default state. Accepts `Light` or `Dark`.
    * @default AvatarVariant.Light
    */
-  variant: AvatarVariant;
+  variant?: AvatarVariant;
   /**
    * The size of the Avatar.
    * @default SystemIconCircleSize.m
    */
-  size: SystemIconCircleSize | number;
+  size?: SystemIconCircleSize | number;
   /**
    * The alt text of the Avatar image.
    * @default Avatar
    */
-  altText: string;
+  altText?: string;
   /**
    * The url of the Avatar image.
    */
   url?: string;
+  /**
+   * The alternative container type for the button. Uses Emotion's special `as` prop.
+   * Will render an `button` tag instead of a `div` when defined.
+   */
+  as?: 'div';
+  /**
+   * The function called when the Avatar is clicked.
+   */
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+  /**
+   * The ref to the button that the styled component renders.
+   */
+  buttonRef?: React.Ref<HTMLButtonElement>;
```

Comment:
I wonder if this should be called `buttonRef` if it could be a div 🤔

Reply:
will rename to elementRef, maybe? In the DropdownButton example it was named buttonRef.

Reply:
I wonder if we should just call it `ref` since we'll be doing that a lot with #548. Any thoughts @NicholasBoll?